### PR TITLE
Migrating and isolating google analytics to tolerate failure to load

### LINKED ISSUE #369: Application list will not load if analytics.js is blocked by adblocker/tracking detection extension
If analytics.js is blocked by any extension, the whole javascript mechanism will not work. What we have to do is to load analytics.js not as a async module via requirejs, but synchronously via a normal script tag. In the code that uses its API, if the API is not there we will have to create a mock object and let it perform requests to google analytics into the void.

### PR DESCRIPTION
**Pull request by stefanoborini**
Fixes #369

### CODE REVIEW

File: remoteappmanager/templates/page.html [hunk 1]
```diff
@@ -41,6 +41,18 @@
...
 
   {% block script %}
+    <script type="text/javascript">
+      // This will always be executed and create the basic ga function that
+      // does nothing
+      window.ga = function () {
+        (window.ga.q = window.ga.q || []).push(arguments);
+      };
+      window.ga.l = 1 * new Date();
+      window.GoogleAnalyticsObject = "ga";
```
Comment (stefanoborini) [inline remoteappmanager/templates/page.html]: There's a problem here, see if you can spot it.

**Comment by stefanoborini** [comment]
@martinRenou There's a small problem in this PR, see if you can spot it

File: jstests/tests/test_analytics.js [hunk 2]
```diff
@@ -4,14 +4,24 @@ define(function(require) {
     
     QUnit.module("Google Analytics");
     QUnit.test("test without analytics", function (assert) {
+        var result=[];
+        window.ga = function(cmd, id, auto) {
+            result[0] = [cmd, id, auto];
+        };
         window.apidata.analytics = undefined;
         var ga = analytics.init();
```
Comment (martinRenou) [inline jstests/tests/test_analytics.js]: You're not using `ga` variable after that, just `analytics.init();` is enough I think.

File: jstests/tests/test_analytics.js [hunk 3]
```diff
@@ -4,14 +4,24 @@ define(function(require) {
...
+        assert.equal(result.length, 0);
     });
     
     QUnit.test("test with analytics", function (assert) {
+        var result=[];
+        window.ga = function(cmd, id, auto) {
+            result[0] = [cmd, id, auto];
+        };
         window.apidata.analytics = {"tracking_id": "X"};
         var ga = analytics.init();
```
Comment (martinRenou) [inline jstests/tests/test_analytics.js]: Same here

File: remoteappmanager/static/js/gamodule.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+], function () {
+    "use strict";
+
+    function init() {
+        var module;
+        
+        if (window.apidata.analytics !== undefined) {
+            window.ga('create', window.apidata.analytics.tracking_id, 'auto');
+        } else {
+            window.ga = function () {
```
Comment (martinRenou) [inline remoteappmanager/static/js/gamodule.js]: I think `window.ga = undefined;` and `window.ga = function () {};` will have the same effect but for the second option jslint would complain about empty brackets

Comment (stefanoborini) [inline remoteappmanager/static/js/gamodule.js]: pycharm complains for the exact opposite reason. It claims return is unnecessary.
by the way, I use jshint, not jslint (jslint is the one by douglas crockford, and while I respect the guy, I think its linter is a bit excessive)

Comment (martinRenou) [inline remoteappmanager/static/js/gamodule.js]: Ahah ok :)

**Pull request merged**